### PR TITLE
Add support for headless usage.

### PR DESCRIPTION
**Pull request by deflaux**
Also add validation to --output which fixes https://github.com/googlegenomics/dataflow-java/issues/31

### CODE REVIEW

**Comment by deflaux** [comment]
@wbrockman please take a look

These changes go with the new task-oriented docs for our dataflow jobs (currently in review) http://googlegenomics.readthedocs.org/en/staging/use_cases/compute_identity_by_state/index.html

**Comment by deflaux** [comment]
I was just reading https://cloud.google.com/compute/docs/gcloud-compute/ which supports `gcloud auth login --no-launch-browser`

Perhaps `--no-launch-browser` might be a better option name here than `--headless`?

**Comment by dionloy** [comment]
SGTM, even if it is poor english =).

**Comment by deflaux** [comment]
@dionloy please take a look

These changes go with the new task-oriented docs for our dataflow jobs (currently in review) http://googlegenomics.readthedocs.org/en/staging/use_cases/compute_identity_by_state/index.html which I just updated to --noLaunchBrowser

File: README.rst [hunk 1]
```diff
@@ -137,11 +137,11 @@ Updating the prebuilt jar
 
 The prebuilt jar is currently generated by hand, whenever the code seems stable and has enough new features to warrant the effort.
 
-To update the jar, we use a different set of maven commands:
+To update the jar, we use a different set of maven commands::
```
Comment (dionloy) [inline README.rst]: :: intentional ?

File: src/main/java/com/google/cloud/genomics/dataflow/utils/GenomicsOptions.java [hunk 2]
```diff
@@ -33,8 +38,15 @@
     public static GenomicsFactory.OfflineAuth getGenomicsAuth(GenomicsOptions options)
         throws IOException, GeneralSecurityException {
       String apiKey = options.getApiKey(), appName = options.getAppName();
-      return GenomicsFactory.builder(appName).setNumberOfRetries(options.getNumberOfRetries()).build()
-          .getOfflineAuth(apiKey, options.getGenomicsSecretsFile());
+      Supplier<? extends VerificationCodeReceiver> verificationCodeReceiver =
+        Suppliers.ofInstance(new GooglePromptReceiver());
```
Comment (dionloy) [inline src/main/java/com/google/cloud/genomics/dataflow/utils/GenomicsOptions.java]: style nit indent 2 more

File: README.rst (hunk 1 above)
Comment (deflaux) [inline README.rst]: yes, its in [reStructured text](http://sphinx-doc.org/rest.html) format

you can see that the format is wrong at the moment here https://github.com/googlegenomics/dataflow-java/blob/master/README.rst#updating-the-prebuilt-jar and this fixes it

**Comment by dionloy** [comment]
LGTM

**Pull request merged**